ddl/placement/rule.go

```diff
@@ -87,6 +104,9 @@ func NewRules(role PeerRoleType, replicas uint64, cnstr string) ([]*Rule, error)
 		ruleCnt := 0
 		for labels, cnt := range constraints2 {
 			if cnt <= 0 {
+				if err := getYamlMapFormatError(string(cnstbytes)); err != nil {
+					return rules, err
```

Comment:
I think it is better to do `return rules, fmt.Errorf("%w: %s", ErrInvalidConstraintsMapcnt, err)`, or even just `return rules, fmt.Errorf("%w:  wrong separator ...", ErrInvalidConstraintsMapcnt)`. We don't need to add a new error for every kind.

Reply:
Done

Reply:
ErrInvalidConstraintsMapcnt is too wordy for users to provide solutions. So "wrong separator ..." is better.

Reply:
Done